iroh-dns-server/src/store/signed_packets.rs

```diff
@@ -110,40 +163,79 @@ impl Actor {
                                 res.send(true).ok();
                             }
                             Message::Remove { key, res } => {
-                                let updated =
-                                    tables.signed_packets.remove(key.as_bytes())?.is_some()
-                                ;
+                                trace!("remove {}", key);
+                                let updated = if let Some(row) = tables.signed_packets.remove(key.as_bytes())? {
+                                    let packet = SignedPacket::from_bytes(&Bytes::copy_from_slice(row.value()))?;
+                                    tables.update_time.remove(&packet.timestamp().to_be_bytes(), key.as_bytes())?;
+                                    inc!(Metrics, store_packets_removed);
+                                    true
+                                } else {
+                                    false
+                                };
                                 if updated {
                                     inc!(Metrics, store_packets_removed);
                                 }
                                 res.send(updated).ok();
                             }
+                            Message::Snapshot { res } => {
+                                trace!("snapshot");
+                                res.send(Snapshot::new(&self.db)?).ok();
+                            }
+                            Message::CheckExpired { key, time } => {
+                                trace!("check expired {} at {}", key, u64::from_be_bytes(time));
+                                tables.update_time.remove_all(&time)?;
+                                if let Some(packet) = get_packet(&tables.signed_packets, &key)? {
+                                    if packet.timestamp() < expired {
+                                        let _ = tables.signed_packets.remove(key.as_bytes())?;
+                                        inc!(Metrics, store_packets_expired);
+                                    }
+                                }
+                            }
                         }
                     }
                 }
             }
             drop(tables);
             transaction.commit()?;
         }
+        Ok(())
     }
 }
 
 /// A struct similar to [`redb::Table`] but for all tables that make up the
 /// signed packet store.
 pub(super) struct Tables<'a> {
     pub signed_packets: redb::Table<'a, &'static SignedPacketsKey, &'static [u8]>,
+    pub update_time: redb::MultimapTable<'a, [u8; 8], SignedPacketsKey>,
 }
 
 impl<'txn> Tables<'txn> {
     pub fn new(tx: &'txn redb::WriteTransaction) -> result::Result<Self, redb::TableError> {
         Ok(Self {
             signed_packets: tx.open_table(SIGNED_PACKETS_TABLE)?,
+            update_time: tx.open_multimap_table(UPDATE_TIME_TABLE)?,
```

Comment:
So, `redb` docs tell me this will create the table, if it doesn't exist already.
No need to wipe the DB & recreate, right? The upgrade should be seamless, right?

Reply:
yes, it is only an issue if you change the schema.